ARCs/arc-0018.md

```diff
@@ -0,0 +1,601 @@
+                    "type": "account"
+                },
+                {
+                    "name": "to",
+                    "type": "account"
+                },
+                {
+                    "name": "current_offered_amount",
+                    "type": "uint64"
+                },
+                {
+                    "name": "current_offered_auth",
+                    "type": "address"
+                }
+            ],
+            "returns": {
+                "type": "void"
+            }
+        },
+        {
+            "name": "get_offer",
+            "desc": "Gets the details offered asset",
+            "readonly": true,
+            "args": [
+                {
+                    "name": "royalty_asset",
+                    "type": "uint64"
+                },
+                {
+                    "name": "from",
+                    "type": "uint64"
+                }
+            ],
+            "returns": {
+                "type": "(address,uint64)"
+            }
+        }
+    ],
+    "locals": [ 
+        {
+            "name": "{uint64 asset id as big endian bytes}",
+            "desc": "The ID of the asset that has been offered as key, authorizing address as value",
+            "type": "(address,uint64)"
+        }  
+    ],
+    "globals": [ 
+        {
+            "name": "royalty_basis",
+            "desc": "The percentage for royalty payment specified in basis points",
+            "type": "uint64"
+        },
+        {
+            "name": "royalty_recipient",
+            "desc": "The receiver of any royalty payment",
+            "type": "address"
+        }
+    ]
+}
+```
+
+#### Example Flow for a Marketplace
+
+```
+Let Alice be the creator of the Royalty Enforcer and Royalty Asset
+Let Alice also be the Royalty Recipient
+Let Bob be the Royalty Asset holder
+Let Carol be a buyer of a Royalty Asset
+```
+
+```mermaid
+sequenceDiagram
+    Alice->>Royalty Enforcer: set_policy with Royalty Basis and Royalty Recipient
+    Alice->>Royalty Enforcer: set_allowed_asset with any asset that should be accepted as payment 
+
+    par List
+        Bob->>Royalty Enforcer: offer
+        Bob->>Marketplace: list
+    end
+
+    Par Buy
+        Carol->>Marketplace: buy
+        Marketplace->>Royalty Enforcer: transfer
+        Bob->>Carol: clawback issued by Royalty Enforcer
+        Royalty Enforcer->>Alice: royalty payment
+    end
+
+    par Delist
+        Bob->>Royalty Enforcer: offer 0 
+        Bob->>Marketplace: delist
+    end
+```
+
+### Metadata
+
+The metadata associated to an asset **MAY** conform to any ARC that supports an additional field in the `properties` section specifying the specific information relevant for off-chain applications like wallets or Marketplace dApps.
+
+The fields that should be specified are:
+
+  - `application_id` - The ID of the Royalty Enforcer application
+  - `application_address` - The Address of the Royalty Enforcer application
```

Comment:
Why do we have these

Reply:
i thought it was just application_id

Reply:
Just convenience for off chain things to make parsing easier. Otherwise its a couple api calls and parsing out encoded bytestrings.

I can remove but they seem useful?